### ISSUE TITLE
which jshint: shows the path to jshint but error from sublime4

### ISSUE THREAD

**Issue by tetoNidan** [SublimeLinter/SublimeLinter-jshint#124]
I installed nvm with the curl method so it exported the environment variables to the .bashrc and I reloaded bash.

I installed jshint with npm:

`$ npm install -g jshint`

    $ which jshint
    /home/tool/.nvm/versions/node/v18.16.0/bin/jshint

my sublimelinter config:

	// SublimeLinter Settings - User
	{
		// "debug": true,
		"lint_mode": "load_save",
		"linters": {

			"pylint": {
				"disable": false,
				"paths": {
					"linux": ["~/.pyenv/shims/"],
				},
				"executable": ["~/.pyenv/shims/pylint"],
			},
			"jshint": {
				"disable": false,
				"paths": {
					"linux": ["~/.nvm/versions/node/v18.16.0/bin/"],
				},
				"executable": ["~/.nvm/versions/node/v18.16.0/bin/jshint"],
			},	
		},
		"show_marks_in_minimap": true,
		"show_hover_line_report": true,
		"styles": [{
			"mark_style": "none",
		}],
	}

Sublime console output when saving:

    SublimeLinter: WARNING: jshint output:
    /usr/bin/env: ‘node’: No such file or directory

pylint is working as expected so I included my entire config file.

What am I doing wrong? Why is sublimelinter looking in /user/bin/env. 

I did a `npm test jshint` and it wanted root permissions? All my node is in local user?  Don't really feel comfortable giving root permissions. Is that necessary?  

Thanks.

**Comment by kaste** [comment]
Here's the revised version of the issue reply:

I'm not entirely sure, but *nvm* may change your environment depending on the current working directory in your shell. Please note that `paths` is not a setting that we support, so you can simply remove that. Perhaps you meant something like this:

```json
"my_linter": {
	"env": {
		"PATH": "~/.nvm/versions/node/v18.16.0/bin/:$PATH"
	}
}
```

Regarding *nvm*, if I recall correctly, the idea was to support it using this pattern: (https://www.sublimelinter.com/en/stable/linter_settings.html#executable)

```json
"executable": ["nvm", "exec", "8.9", "eslint"]
```

In other words, we delegate to *nvm* just like we execute `yarn run eslint` if we detect a yarn-powered project.

Whether *nvm* is actually ready to use within Sublime Text depends on how you start Sublime Text as well. If I remember correctly, the main scripts of *nvm* run on interactive login shells. If that's correct, starting Sublime Text from the shell (using `subl .`) might be necessary.

I believe the upstream issue at https://github.com/SublimeLinter/SublimeLinter/issues/1795 is the one that will resolve your use case if it gets implemented.

**Comment by tetoNidan** [comment]
>I'm not entirely sure, but nvm may change your environment depending on the current working directory in your shell.

That would make sense. I was thinking that might be the case because I could only install jshint with the -g flag when I was in my home directory. I guess when I installed it without the -g flag I was in the correct directory and didn't know how things worked and got lucky! 

Looks like Im going to have to fine tooth comb the documentation.

Thanks, That clears up what I needed to know. Looks like I was way off on the `executable`! I will close this and test.
